core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -302,6 +305,98 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   }
 
   /**
+   * testSendBeforeAndAfterPartitionExpansion checks the partitioning behavior before and after partitions are added
+   *
+   * The specified partition-ids should be respected
```

Comment:
Maybe this could be: "The producer will successfully send messages to the partition specified in each record"